src/main/java/it/pagopa/pm/gateway/controller/XPayPaymentController.java

```diff
@@ -227,15 +229,43 @@ private ResponseEntity<XPayAuthPollingResponse> createXPayAuthPollingResponse(Ht
 
         String requestId = entity.getGuid();
         log.info("START - create XPay polling response for requestId: " + requestId);
-        response.setStatus(entity.getStatus());
-        response.setHtml(entity.getXpayHtml());
+        String status = entity.getStatus();
+        log.info(String.format("Request status for requestId %s is %s", requestId, status));
+
+        response.setStatus(status);
+        switch (getEnumValueFromString(status)) {
+            case CREATED:
+                String xpayHtml = entity.getXpayHtml();
+                response.setHtml(xpayHtml);
+                if (StringUtils.isBlank(xpayHtml)) {
+                    log.info(String.format("HTML from XPay for requestId %s has not been acquired yet", requestId));
+                }
+                if (Objects.isNull(entity.getAuthorizationOutcome())) {
+                    log.info(String.format("Authorization for requestId %s is pending", requestId));
+                    response.setAuthOutcome(PENDING_MSG);
```

Comment:
to be reviewed, everything must be clear from the status without this custom outcome

Reply:
Since this status is not actually useful to XPay logic, I thought to remove it